app/main.py

```diff
@@ -1,8 +1,23 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    some_list = []
+    for person_data in people:
+        some_list.append(
+            Person(name=person_data["name"], age=person_data["age"]))
+
+    for person_data in people:
+        person = Person.people[person_data["name"]]
+        spouse_key = "wife" if "wife" in person_data else "husband"
+        spouse_name = person_data.get(spouse_key)
+        if spouse_name:
+            setattr(person, spouse_key, Person.people[spouse_name])
```

Comment:
The logic for setting a spouse attribute assumes that the spouse's name is always present in the `Person.people` dictionary. If the spouse's name is not found, this will raise a `KeyError`. Consider adding error handling to manage such cases.